tests/bootstrap.php

```diff
@@ -1,19 +1,36 @@
 <?php
 
-$projectRoot = dirname(__DIR__);
+$projectRootDirectory = dirname(__DIR__);
+$projectConfigDirectory = $projectRootDirectory . '/tests/config';
+$casserverModulePath = $projectRootDirectory . '/vendor/simplesamlphp/simplesamlphp/modules/casserver';
+$simplesamlphpConfig = $projectRootDirectory . '/vendor/simplesamlphp/simplesamlphp/config';
+
 /** @psalm-suppress UnresolvableInclude */
-require_once($projectRoot.'/vendor/autoload.php');
+require_once($projectRootDirectory . '/vendor/autoload.php');
 
-// Symlink module into ssp vendor lib so that templates and urls can resolve correctly
-$linkPath = $projectRoot.'/vendor/simplesamlphp/simplesamlphp/modules/casserver';
-if (file_exists($linkPath) === false) {
-    print "Linking '$linkPath' to '$projectRoot'\n";
-    symlink($projectRoot,$linkPath);
-} else {
-    if (is_link($linkPath) === false) {
-        // Looks like the pre-installed casserver module is here. Lets remove it and symlink in this one
-        print "Renaming pre-installed casserver module and linking '$linkPath' to '$projectRoot'\n";
-        rename($linkPath, $linkPath.'-preinstalled');
-        symlink($projectRoot, $linkPath);
+/**
+ * Sets a link in the simplesamlphp vendor directory
+ * @param $target
+ * @param $link
+ */
+function symlinkModulePathInVendorDirectory($target, $link)
+{
+    if (file_exists($link) === false) {
+        // If the link is invalid, remove it.
```

Comment:
You mean `valid` instead?

Reply:
Step 1: check if a file exists, if not, continue:     if (file_exists($link) === false) {
When you enter an invalid link, the file does not exist. But link can still be there.

Step 2: if is an invalid link,  remove it. So the comment is about the next line.

Step 3: Place new link.